### ISSUE TITLE
Provide JSON validation for config files

### ISSUE THREAD

**Issue by MahdiBM** [swiftlang/vscode-swift#1208]
**Is your feature request related to a problem? Please describe.**
This extension should provide [JSON schema](https://json-schema.org/draft-07)s for validation, for JSON config files related to this extension.
For example for `.sourcekit-lsp/config.json`, [when sourcekit-lsp supports that](https://github.com/swiftlang/sourcekit-lsp/issues/1829).
Then users will have an easier time writing config files.

Please Note that VS Code does not fully support newer version of JSON schema specification and we'll probably need to stick to version `draft-07`.

For the JSON validations to be enabled, this extension can use the `package.json` file. See [rust-analyzer](https://github.com/rust-lang/rust-analyzer/blob/9224ec44978b6817510b0b2999c7cf25c1ab837f/editors/code/package.json#L3212C10-L3212C24) as an example.

**Describe alternatives you've considered**
Users can manually use [`"$schema": "link/to/schema"`](https://code.visualstudio.com/Docs/languages/json#_json-schemas-and-settings) in the config JSON files to manually enable schema validations. See the link for a full of the options.

**Additional context**

**Comment by award999** [comment]
With the work done by https://github.com/swiftlang/sourcekit-lsp/pull/1849, this should be straightforward and helpful. The unfortunate thing is that you can only statically set this file as a manifest contribution.
```
 "jsonValidation": [
      {
        "fileMatch": "**/.sourcekit-lsp/config.json",
        "url": "https://raw.githubusercontent.com/swiftlang/sourcekit-lsp/refs/heads/main/config.schema.json"
      }
    ],
```

This means we cannot set it based on the sourcekit-LSP version that is actually being used. Dynamically setting this has been requested issue against VSCode, but doesn't have a lot of traction by the looks of it: https://github.com/microsoft/vscode/issues/230136

So since we're stuck with static, there is then a couple possible issues:

1. Do we pin to "main" or to the latest stable release version of the schema? The pro for main is we don't have to update it for each release. We also have no idea which version they may be using (could be old) so we could let them see all the new configurations as a slight incentive to upgrade toolchain versions. The pro for latest release version is maybe we don't want to push people to main, although it has only been updated once since 6.1.
2. Do we bundle the schema statically in the extension, or specify a Github URL like above? Static is always there but we'd then have to update the file manually for new schema versions. Using the URL you dynamically you get updates for free, but may hit a GH outage.

Also worth nothing that in the .sourcekit-lsp/config.json the user could override the schema, pinning it to a specific release, so even though we'd be statically pinning, the user could override.
```
{
    "$schema": "https://raw.githubusercontent.com/swiftlang/sourcekit-lsp/refs/heads/release/6.1/config.schema.json",
    "backgroundIndexing": true,
    "logging": {"level": "info"}
}
```

Another thought is we could add a `Generate Sourcekit-LSP configuration file` command, that adds that `$schema` property based on the toolchain they're using.

There is of course the option of just letting this be a manual task for the user, but there is some manual digging and steps they have to do and may not even be aware that is an option. I think the tradeoffs are worth having something working for users out of the box.

**Comment by award999** [comment]
@MahdiBM in writing all that I've leaned towards some `Generate Sourcekit-LSP configuration file` command as I myself always have to lookup the file name and path that I'm supposed to create the file at. This way we can add the `$schema` with the right toolchain version. Maybe we even check the file when it's open and prompt the user to update the `$schema` if the toolchain version does not match.

I can still add this as an all else fails fall back
```
"jsonValidation": [
      {
        "fileMatch": "**/.sourcekit-lsp/config.json",
        "url": "https://raw.githubusercontent.com/swiftlang/sourcekit-lsp/refs/heads/main/config.schema.json"
      }
    ],
```

does that work for your needs?

**Comment by MahdiBM** [comment]
Does VSCode allow setting the schema to a  local file?
Then we'd be able to update that file based on sourcekit-lsp version.

**Comment by award999** [comment]
Nope, it is completely static, has to be a static URL or path within the bundled extension

**Comment by MahdiBM** [comment]
Hmmm. I guess the command is good to have, adds a bit of visibility too. If there is a soucekitlsp config file and it has incorrect or missing $schema based on the current active toolchain, the extension can provide a quick-fix/Fix-It for it hopefully or at least let the user know.

If there can't be a quick-fix/Fix-It and it becomes non-trivual, then we might also need an option to disable the check/suggestion so users can disable it if they're switching a lot between toolchains or something. If there can be a quick-fix then it should be fine with no disable option, we can add it later on if so eone is annoyed.

That's my opinion so far.

**Comment by rbenegal** [comment]
Confirmed with 6955cbee that the `Generate Sourcekit-LSP configuration file` works well for workspaces and single folder projects. 

Note: Modifying the config.json file currently doesn't prompt the user for a LSP restart but I think it should so I will make a new issue for this.
